services/impl/src/main/java/org/exoplatform/automatic/translation/rest/AutomaticTranslationRestService.java

```diff
@@ -118,27 +93,44 @@ public Response setActiveConnector(@ApiParam(value = "Connector name", required
   @Path("/setApiKey")
   @RolesAllowed("administrators")
   @Consumes(MediaType.APPLICATION_JSON)
-  @ApiOperation(
-      value = "Set Automatic translation API key",
-      httpMethod = "PUT",
-      response = Response.class,
-      produces = MediaType.APPLICATION_JSON
-  )
-  @ApiResponses(
-      value = {
-          @ApiResponse(code = 200, message = "Request fulfilled"),
-          @ApiResponse(code = 403, message = "Not authorized to change the connector")
-      }
-  )
+  @ApiOperation(value = "Set Automatic translation API key", httpMethod = "PUT", response = Response.class, produces = MediaType.APPLICATION_JSON)
+  @ApiResponses(value = { @ApiResponse(code = 200, message = "Request fulfilled"),
+      @ApiResponse(code = 403, message = "Not authorized to change the connector") })
   public Response setApiKey(@ApiParam(value = "Connector name", required = true) @QueryParam("connector") String connector,
                             @ApiParam(value = "Api Key", required = true) @QueryParam("apikey") String apikey) {
     connector = (connector == null || connector.equals("null")) ? "" : connector;
-    if (automaticTranslationService.setApiKey(connector,apikey)) {
+    if (automaticTranslationService.setApiKey(connector, apikey)) {
       return Response.ok().build();
     } else {
       return Response.status(403).build();
```

Comment:
The response code is wrong, in fact the user is allowed to set the API Key but the connector plugin doesn't exist, so the response code should be 400 : Bad request (with a detailed message as Response.entity to be able it in UI by example).

Reply:
done